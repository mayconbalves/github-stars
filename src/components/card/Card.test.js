import { render, screen } from '@testing-library/react'
import Card from './'

describe('<Card />', () => {
  let props = {
    cloneUrl: 'https://github.com/mayconbalves/app-go-barber.git',
    license: {
      name: 'MIT license'
    },
    stars: 1,
    title: 'app go barber'
  }

  it('should render first title', () => {
    const { container } = render(<Card {...props} />)
    expect(screen.getByText(/Clone repo:/i)).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })
  it('should render subtitle', () => {
    const { container } = render(<Card {...props} />)
    expect(screen.getByText(/License type:/i)).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })
})
