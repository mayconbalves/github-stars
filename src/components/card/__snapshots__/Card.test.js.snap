// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> should render correct empty img 1`] = `
<div
  class="sc-bdfBwQ kpFlCc"
>
  <div
    class="sc-gsTCUz eptAzo"
  >
    <div
      class="sc-dlfnbm dSCSBh"
    >
      <p>
        app go barber
      </p>
    </div>
    <div
      class="sc-hKgILt jbBhkH"
    >
      <p>
        1
      </p>
      <img
        alt="full star"
        class="sc-eCssSg bTrOV"
        src="full_star.png"
      />
    </div>
  </div>
  <div
    class="sc-jSgupP cXdENn"
  >
    <ul>
      <li
        class="sc-gKsewC gIrsof"
      >
        <h1
          class="sc-iBPRYJ fgpkrE"
        >
          Clone repo:
        </h1>
        <p
          class="sc-fubCfw eulNsP"
        >
          https://github.com/mayconbalves/app-go-barber.git
        </p>
      </li>
      <li
        class="sc-gKsewC gIrsof"
      >
        <h1
          class="sc-iBPRYJ fgpkrE"
        >
          License type:
        </h1>
        <p
          class="sc-fubCfw eulNsP"
        >
          None
        </p>
      </li>
    </ul>
  </div>
</div>
`;

exports[`<Card /> should render first title 1`] = `
<div
  class="sc-bdfBwQ kpFlCc"
>
  <div
    class="sc-gsTCUz eptAzo"
  >
    <div
      class="sc-dlfnbm dSCSBh"
    >
      <p>
        app go barber
      </p>
    </div>
    <div
      class="sc-hKgILt jbBhkH"
    >
      <p>
        1
      </p>
      <img
        alt="full star"
        class="sc-eCssSg bTrOV"
        src="full_star.png"
      />
    </div>
  </div>
  <div
    class="sc-jSgupP cXdENn"
  >
    <ul>
      <li
        class="sc-gKsewC gIrsof"
      >
        <h1
          class="sc-iBPRYJ fgpkrE"
        >
          Clone repo:
        </h1>
        <p
          class="sc-fubCfw eulNsP"
        >
          https://github.com/mayconbalves/app-go-barber.git
        </p>
      </li>
      <li
        class="sc-gKsewC gIrsof"
      >
        <h1
          class="sc-iBPRYJ fgpkrE"
        >
          License type:
        </h1>
        <p
          class="sc-fubCfw eulNsP"
        >
          None
        </p>
      </li>
    </ul>
  </div>
</div>
`;

exports[`<Card /> should render subtitle 1`] = `
<div
  class="sc-bdfBwQ kpFlCc"
>
  <div
    class="sc-gsTCUz eptAzo"
  >
    <div
      class="sc-dlfnbm dSCSBh"
    >
      <p>
        app go barber
      </p>
    </div>
    <div
      class="sc-hKgILt jbBhkH"
    >
      <p>
        1
      </p>
      <img
        alt="full star"
        class="sc-eCssSg bTrOV"
        src="full_star.png"
      />
    </div>
  </div>
  <div
    class="sc-jSgupP cXdENn"
  >
    <ul>
      <li
        class="sc-gKsewC gIrsof"
      >
        <h1
          class="sc-iBPRYJ fgpkrE"
        >
          Clone repo:
        </h1>
        <p
          class="sc-fubCfw eulNsP"
        >
          https://github.com/mayconbalves/app-go-barber.git
        </p>
      </li>
      <li
        class="sc-gKsewC gIrsof"
      >
        <h1
          class="sc-iBPRYJ fgpkrE"
        >
          License type:
        </h1>
        <p
          class="sc-fubCfw eulNsP"
        >
          None
        </p>
      </li>
    </ul>
  </div>
</div>
`;
